### PR TITLE
[SPARK-30963][INFRA] Add GitHub Action job for document generation

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to add a new `GitHub Action` job for document generation.

### Why are the changes needed?

We had better test the document generation in PR Builder.
- https://lists.apache.org/thread.html/rd06a2154e853812652b8f7fa3c003746ed531b213c531517f055e1dc%40%3Cdev.spark.apache.org%3E

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Pass the GitHub Action in this PR.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @HyukjinKwon and @shaneknapp

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -117,3 +117,41 @@ jobs:
...
+        architecture: 'x64'
+    - uses: actions/setup-ruby@v1
+      with:
+        ruby-version: '2.7'
+    - name: Install R
+      run: |
+        echo 'deb https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/' | sudo tee -a /etc/apt/sources.list
+        curl -sL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0xE298A3A825C0D65DFD57CBB651716619E084DAB9" | sudo apt-key add
+        sudo apt-get update
+        sudo apt-get install -y r-base r-base-dev libcurl4-openssl-dev
```
Comment (viirya) [inline .github/workflows/master.yml]: Do we need pandoc for R document? I saw:

> If you'd like to generate R documentation, you'll need to [install Pandoc](https://pandoc.org/installing.html) and install these libraries ...

Comment (dongjoon-hyun) [reply .github/workflows/master.yml]: Oh, thanks. Let me check that too, @viirya !

**Comment by HyukjinKwon** [comment]
Oh nice. +1. Let me take a look today!

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon ! Please let me know if I missed something.

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -117,3 +117,40 @@ jobs:
...
+      run: |
+        echo 'deb https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/' | sudo tee -a /etc/apt/sources.list
+        curl -sL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0xE298A3A825C0D65DFD57CBB651716619E084DAB9" | sudo apt-key add
+        sudo apt-get update
+        sudo apt-get install -y r-base r-base-dev libcurl4-openssl-dev pandoc
+    - name: Install packages
+      run: |
+        pip install sphinx mkdocs numpy
+        gem install jekyll jekyll-redirect-from rouge
+        sudo Rscript -e "install.packages(c('curl', 'xml2', 'httr', 'devtools', 'testthat', 'knitr', 'rmarkdown', 'roxygen2', 'e1071', 'survival'), repos='https://cloud.r-project.org/')"
```
Comment (HyukjinKwon) [inline .github/workflows/master.yml]: @dongjoon-hyun, out of curiosity, do you remember why we needed `curl`, `xml2`, `httr` when we add `lint-r` job?

I thought it should be as below (from `docs/REAME.md`)

```bash
sudo Rscript -e 'install.packages(c("knitr", "devtools", "testthat", "rmarkdown", "roxygen2"), repos="https://cloud.r-project.org/")'
```

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise.

**Review by viirya — APPROVED** [review]

File: .github/workflows/master.yml (hunk 2 above)
Comment (dongjoon-hyun) [reply .github/workflows/master.yml]: IIRC, it was the legacy migrated from the old Jenkins jobs when we migrate that from old Jenkins. I believe we can remove them later if the test passed.

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon and @viirya . Since this is the first attempt to recover this test coverage, we can improve gradually from now.
Merged to master/3.0.

**Pull request closed without merging**